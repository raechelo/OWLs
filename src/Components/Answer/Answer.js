import React, { Component } from 'react';


class Answer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      allAnswers: [],
      correctAnswer: '',
      userAnswer: ''
    }
  }

  updateAnswers = () => {
    this.setState( {
      allAnswers: this.props.allAnswers,
      correctAnswer: this.props.correctAnswer
    } )
  }



  render() {
    let btns = this.state.allAnswers.sort(() => 0.5 - Math.random()).map(a => {
      return (<button value={a}>a</button>)
    })
    return (btns)
  }
}


export default Answer;