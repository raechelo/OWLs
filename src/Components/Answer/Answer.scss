.ans-btn {
  background-color: hsla(37, 70%, 80%, 1);
  border: none;
  color: $board-text;
  cursor: pointer;
  font-family: $bodyfont;
  font-size: .75rem;
  padding:  15px 20px;
  margin: .5rem 1rem;
  text-align: center;
}

.ans-btn-section {
  display: grid;
  grid-template-columns: 2fr 2fr;
  margin: 1rem auto;

  button {
    &:hover {
      box-shadow: 2px 2px 3px $boardBg-shadow;
      background-color: $board-background-hover;
      color: $board-btn-text;
      text-decoration: underline;
    }
  }
}