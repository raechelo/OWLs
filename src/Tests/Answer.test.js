import React from 'react';
import Answer from '../Components/Answer/Answer';
import { shallow } from 'enzyme';

describe('Answer', () => {

  let wrapper;

  const mockAnswers = ['Moony', 'Wormtail', 'Padfoot', 'Prongs']

  const mockCorrectAnswer = 'Padfoot';

  const mockNextQuestion = jest.fn();

  const mockIncorrectAns = jest.fn(); 

  beforeEach(() => {
    wrapper = shallow (
      <Answer allAnswers={mockAnswers} correctAnswer={mockCorrectAnswer} incorrectAns={mockIncorrectAns} nextQuestion={mockNextQuestion} />
    )
  });

  it.skip('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual( {allAnswers: [], correctAnswer: '', userAns: ''} );
  });

  it('should update state', () => {
    expect(wrapper.state()).toEqual( {allAnswers: [], correctAnswer: '', userAns: ''} );
    wrapper.instance().updateState( { target: { value: 'Caput Draconis' } } );
    expect(wrapper.state()).toEqual( {allAnswers: mockAnswers, correctAnswer: mockCorrectAnswer, userAns: 'Caput Draconis'} );
  });

  
})