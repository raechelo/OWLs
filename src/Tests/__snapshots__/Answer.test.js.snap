// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should match the snapshot with all the data passed in 1`] = `undefined`;
